### PR TITLE
Update docker-py

### PR DESCRIPTION
**Pull request by yongtang**
This fix update docker-py so that containers from the tests run
could be cleaned up during teardown:
```diff
-ENV DOCKER_PY_COMMIT 5e28dcaace5f7b70cbe44c313b7a3b288fa38916
+ENV DOCKER_PY_COMMIT 8b246db271a85d6541dc458838627e89c683e42f
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by yongtang** [comment]
This PR depends on https://github.com/docker/docker-py/pull/1922

File: Dockerfile
Comment (thaJeztah) [inline Dockerfile]: ^^ commenting just so that we don't forget to change it before merging

**Review by thaJeztah — CHANGES_REQUESTED** [review]

**Comment by thaJeztah** [comment]
https://github.com/docker/docker-py/pull/1922 was merged 👍

**Comment by yongtang** [comment]
Thanks @thaJeztah. The PR has been updated.

**Review by thaJeztah — APPROVED** [review]
LGTM (if green)

**Comment by thaJeztah** [comment]
ping @vdemeester @cpuguy83 it's green 🤗

**Review by boaz0 — APPROVED** [review]
LGTM :+1:

**Review by vdemeester — APPROVED** [review]
LGTM 🦌

**Pull request merged**
